nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -535,6 +537,8 @@ def build_transformer_config(self) -> TransformerConfig:
             'recompute_granularity': recompute_granularity,
             'recompute_method': recompute_method,
             'recompute_num_layers': recompute_num_layers,
+            'calculate_per_token_loss': calculate_per_token_loss,
+            'finalize_model_grads_func': self.finalize_model_grads,
```

Comment:
@yaoyu-33 here you need to be careful, to only set it if the user is not using Mcore's distOpt, for example https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py#L676